src/hardware/siglent-sds/protocol.c

```diff
@@ -75,10 +88,11 @@ static int siglent_sds_event_wait(const struct sr_dev_inst *sdi)
 			if (sr_scpi_get_string(sdi->conn, ":INR?", &buf) != SR_OK)
 				return SR_ERR;
 			sr_atoi(buf, &out);
+			g_free(buf);
 			g_usleep(s);
-		} while (out == 0);
-
-		sr_dbg("Device triggered.");
+		} while ((out & 1) != 1);
```

Comment:
The old code ends the loop and claims "triggered" even if the value is `8192` which means "Trigger is ready". Check the first bit only which stands for "New signal acquired"